File: attendance-tracker-react/src/services/userService.js
import api from "./api";

export const userService = {
  getUsers: () => api.get("/users"),
  getUserById: (userId) => api.get(`/users/${userId}`),
  updateFaceEmbedding: (userId, faceEmbeddingJson) =>
    api.patch(
      `/users/${userId}?faceEmbeddingJson=${encodeURIComponent(
        faceEmbeddingJson
      )}`
    ),
};

File: attendance-tracker-springboot/src/main/java/com/trainee_project/attendance_tracker_springboot/service/UserService.java
package com.trainee_project.attendance_tracker_springboot.service;

import com.trainee_project.attendance_tracker_springboot.dto.UserResponseDto;

import java.util.List;

public interface UserService {
    List<UserResponseDto> getUserList(String email);

    UserResponseDto getUserById(String userId);

    UserResponseDto updateFaceEmbeddingJson(String userId, String faceEmbeddingJson);

    UserResponseDto getUserProfile(String email);
}
